main.go

```diff
@@ -1,5 +1,67 @@
 package main
 
+import (
+	"context"
+	"log"
+
+	"github.com/red-hat-storage/managed-fusion-fleet-reconciler/pkg/config"
+	"github.com/red-hat-storage/managed-fusion-fleet-reconciler/pkg/db"
+	"github.com/red-hat-storage/managed-fusion-fleet-reconciler/pkg/forman"
+	"github.com/red-hat-storage/managed-fusion-fleet-reconciler/pkg/reconciler"
+	"github.com/red-hat-storage/managed-fusion-fleet-reconciler/pkg/types"
+
+	"go.uber.org/zap"
+)
 
 func main() {
+	// parse configuration from yaml file
+	conf, err := config.GetConfig("config/config.yaml")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	logger, err := zap.NewProduction()
+	if conf.General.Development {
+		logger, err = zap.NewDevelopment()
+	}
+	if err != nil {
+		log.Fatal(err)
+	}
+	ctx := context.Background()
+	dbInstance, err := db.NewPGInstance(ctx, logger, &conf.DB)
+	if err != nil {
+		logger.Fatal(err.Error())
+	}
+
+	if err := dbInstance.Ping(ctx); err != nil {
+		logger.Fatal(err.Error())
+	}
+
+	reqChan := forman.GoForman(logger, conf.Worker.Count, reconciler.Reconcile)
+
+	pgConn, err := dbInstance.SetupNotification(ctx, reqChan)
+	if err != nil {
+		logger.Fatal(err.Error())
+	}
+
+	defer pgConn.Close(ctx)
+
+	providers, err := dbInstance.ListProviderClusters(ctx)
+	if err != nil {
+		logger.Fatal(err.Error())
+	}
+
+	go func(providers []*types.ProviderCluster) {
+		for _, p := range providers {
+			req := types.Request{}
+			req.Name = p.ClusterID
+			reqChan <- req
+		}
+	}(providers)
+
+	for {
+		if err := pgConn.WaitForNotification(ctx); err != nil {
```

Comment:
- referring to the [doc](https://github.com/jackc/pgx/blob/master/doc.go#L168) provided in another comment what are we doing with the notification and doesn't this method return two values?
- or we'll get only error due to already setting up notification earlier?

Reply:
its a wrapper over the pgconn.Notification here we are directly using pgconn and settling the Handler, if we go with this one we need to write our own handler to process the data